src/GitVersionCore/Helpers/LibGitExtensions.cs

```diff
@@ -138,5 +138,22 @@ public static void DumpGraph(string workingDirectory, Action<string> writer = nu
                 Console.Write(output.ToString());
             }
         }
+
+        public static bool IsBranch(this string branchName, string branchNameToCompareAgainst)
```

Comment:
I think extension methods like these are a clear indication that we need a `Branch` class with an `IEqualityComparer<Branch>` implementation. Perhaps not as a part of this PR, but some day.

Reply:
I would agree, in another PR